packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx

```diff
@@ -66,81 +68,95 @@ export function PresentationInstanceFilterDialog(props: PresentationInstanceFilt
       <Dialog.Backdrop />
       <Dialog.Main className="presentation-instance-filter-dialog-content-container">
         <Dialog.TitleBar className="presentation-instance-filter-title" titleText={title ? title : translate("instance-filter-builder.filter")} />
-        <Dialog.Content className="presentation-instance-filter-content">
-          {descriptor instanceof Descriptor ? (
-            <PresentationInstanceFilterBuilder {...restProps} descriptor={descriptor} onInstanceFilterChanged={onInstanceFilterChanged} />
-          ) : (
-            <DelayLoadedPresentationInstanceFilterBuilder {...restProps} onInstanceFilterChanged={onInstanceFilterChanged} descriptorGetter={descriptor} />
-          )}
-        </Dialog.Content>
-        <div className="presentation-instance-filter-dialog-bottom-container">
-          <div>{filterResultCountRenderer && filterResultCountRenderer(filter)}</div>
-          <Dialog.ButtonBar className="presentation-instance-filter-button-bar">
-            <Button className="presentation-instance-filter-dialog-apply-button" styleType="high-visibility" onClick={applyButtonHandle} disabled={!filter}>
-              {translate("instance-filter-builder.apply")}
-            </Button>
-            <Button className="presentation-instance-filter-dialog-close-button" onClick={onClose}>
-              {translate("instance-filter-builder.cancel")}
-            </Button>
-          </Dialog.ButtonBar>
-        </div>
+        {descriptor instanceof Descriptor ? (
+          <PresentationInstanceFilterDialogContent {...restProps} descriptor={descriptor} onClose={onClose} />
+        ) : (
+          <DelayedCenteredProgressRadial />
+        )}
       </Dialog.Main>
     </Dialog>
   );
 }
 
-interface DelayLoadedPresentationInstanceFilterBuilderProps extends Omit<PresentationInstanceFilterBuilderProps, "descriptor"> {
-  descriptorGetter: () => Promise<Descriptor>;
-}
-
-function DelayLoadedPresentationInstanceFilterBuilder(props: DelayLoadedPresentationInstanceFilterBuilderProps) {
-  const { descriptorGetter, ...restProps } = props;
-  const descriptor = useDelayLoadedDescriptor(descriptorGetter);
-  if (!descriptor) {
-    return <DelayedCenteredProgressRadial />;
-  }
-
-  return <PresentationInstanceFilterBuilder {...restProps} descriptor={descriptor} />;
-}
-
-function useDelayLoadedDescriptor(descriptorGetter: () => Promise<Descriptor>) {
-  const [descriptor, setDescriptor] = useState<Descriptor>();
+function useDelayLoadedDescriptor(descr: Descriptor | (() => Promise<Descriptor>)) {
+  const [descriptor, setDescriptor] = useState<Descriptor | (() => Promise<Descriptor>)>(() => descr);
+  const disposed = useRef<boolean>(false);
```

Comment:
You could use local variable scoped to `useEffect` for tracking status of disposal.

Reply:
fixed